Add unit tests GA workflow, refs #13460

jraddaoui opened the pull request:
Adds a Github Actions workflow to run the existing unit tests over
Ubuntu 18.04 and a matrix of PHP versions. These tests require access
to MySQL/Percona and Elasticsearch, which are setup using the Docker
Compose environment (only those two services). The workflow will be
triggered on all PRs and new commits on qa/* and stable/* branches.

djjuhasz commented on .github/workflows/unit-tests.yml:
```diff
@@ -0,0 +1,48 @@
+    strategy:
+      fail-fast: false
+      matrix:
+        php: [7.2, 7.3, 7.4, 8.0]
```
Let's remove the PHP 8.0 test until we have time to update the code so the tests pass.

djjuhasz commented on test/bootstrap/config.php:
```diff
@@ -0,0 +1,70 @@
+        'pooling' => true,
+        'dsn' => 'mysql:host=127.0.0.1;port=63003;dbname=atom;charset=utf8mb4',
+        'username' => 'atom',
+        'password' => 'atom_12345',
```
Is having the password in plain text a security risk?

jraddaoui replied:
I don't think it is ...

It's just an initial database for the tests and we already have that password in the code as default for the development env. Also the VM is up only for a few minutes and I guess it's not accessible from the outside and even if it's, the Percona port is only accessible internally:

https://github.com/artefactual/atom/blob/qa/2.x/docker/docker-compose.dev.yml#L57

djjuhasz reviewed (changes requested):
Looks good to me @jraddaoui.  :)  I did request that we drop the PHP 8.0 test until we can get it to pass.

jraddaoui committed:
Add unit tests GA workflow, refs #13460

Adds a Github Actions workflow to run the existing unit tests over
Ubuntu 18.04 and a matrix of PHP versions. These tests require access
to MySQL/Percona and Elasticsearch, which are setup using the Docker
Compose environment (only those two services). The workflow will be
triggered on all PRs and new commits on qa/* and stable/* branches.
Apart from the unit tests, this workflow also tests the database and
Elasticsearch index initialization using the purge task.

jraddaoui commented:
It's worth mention that, apart from the unit tests, this workflow also tests the database and Elasticsearch index initialization using the purge task ...

https://github.com/artefactual/atom/blob/dde852f601ff620e47120a7608fb2e7f50a3b9f9/.github/workflows/unit-tests.yml#L45-L46

djjuhasz reviewed (approved):
Looks great! Thanks @jraddaoui :shipit:

jraddaoui commented:
c89065a in stable/2.6.x.
